src/mon/MgrMap.h

```diff
@@ -514,13 +539,18 @@ class MgrMap
       }
       f->close_section();
     }
+    f->close_section(); // always_on_modules
     f->dump_int("last_failure_osd_epoch", last_failure_osd_epoch);
     f->open_array_section("active_clients");
-    for (const auto &c : clients) {
-      f->dump_object("client", c);
+    for (const auto &i: clients) {
+      f->open_object_section("client");
+      if (!i.first.empty()) {
+	f->dump_string("name", i.first);
```

Comment:
IMO `name` should be present even if empty -- so that the structure of the output is always the same.